### DISCUSSION TITLE
Some questions about time format and dataframe

### DISCUSSION THREAD

**Discussion by Protomolekule** [highfestiva/finplot#492] [General]
I will start with the simple questions:

** What is the perfect dataframe for finplot?** 
I've seen examples using datetime index and datetime column. Please show me df.info() and some rows. 
Does the df DateTime index need to be offset-naive or offset-aware?

**How to handle TimeZone and localisation?**
Do I have to remove localisation before plotting?

**How to set DateTime format for the crosshair info?**

The crosshair is build in __init__:
`def _x2local_t(datasrc, x):
if display_timezone == None:
        return _x2utc(datasrc, x)
    return _x2t(datasrc, x, lambda t: _millisecond_tz_wrap(datetime.fromtimestamp(t/1e9, tz=display_timezone).strftime(timestamp_format)))
`
The timestamp_format is set also in the __init__:
`timestamp_format = '%Y-%m-%d %H:%M:%S.%f'`

but in the crosshair only up to the hrs will be displayed (in a extra row without :00). I want to have it like '29.01.24\n10:24' No idea how this will build.

The solution mentioned in the [finplot/examples/snp500.py](url) is without manipulating the time format.

Thanks for your attention

**Comment by highfestiva** [comment]
Either offset naive or aware is fine. If you wish to change the display timezone to something other than that of your computer's, you should set `finplot.display_timezone`. See separate questions, wiki, snippets, etc. on that.

Try setting the `finplot.timestamp_format` to `'%d.%m.%y\n%H:%M'` and see what happens.

**Reply by Protomolekule** [reply]
Hm, nothing special. I've played around with different combinations, but the minutes of  `finplot.timestamp_format` are ignored...

Regarding Timezone: Depending on provider there are different time formats depending on candlesize etc. So I run into problems using their data. So, I came over to the idea to implement a function which normalizes the incoming datetimes to one format I will use on every chart, calculation etc. 
Somtimes I see timeformat in finplot as integer and datetime with timezone etc. Therefore I'm asking about the prefered datetime format.
